### PR TITLE
"Don't directly throw Exception" System.Data.SqlClient

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
@saurabh500 i've found other place where there are "throw new Exception()" what do you think?Instead of ArgumentException i found also a more specialized exception InvalidUdtException(but i'm not sure if make sense). 
contributes to #23748

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SSRP.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@ internal static int GetPortByInstanceName(string browserHostName, string instanc
             }
             catch (SocketException se)
             {
-                throw new Exception(SQLMessage.SqlServerBrowserNotAccessible(), se);
+                throw new System.InvalidOperationException(SQLMessage.SqlServerBrowserNotAccessible(), se);
```
Comment (danmosemsft) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SSRP.cs]: generally one has `using System;` and do not prefix with System

**Comment by saurabh500** [comment]
Cc @corivera for back compat check and review

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SSRP.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SSRP.cs]: yes sorry

**Comment by saurabh500** [comment]
@MarcoRossignoli have you evaluated if these changes can change the exceptions being surfaced to the user?

**Comment by MarcoRossignoli** [comment]
@saurabh500 actually no(only run current tests), if you want i can try to create a "test" for every case to understand, do you agree?

**Comment by stephentoub** [comment]
What is the value of the changes being made here?  If the exceptions are only internal, then the type doesn't really matter, and if the exceptions would bubble out, it sounds like we'd be concerned about the visibility of those changes and wouldn't want to take the change.

**Comment by MarcoRossignoli** [comment]
@stephentoub @corivera @saurabh500  if that'd be useful i can analyse if there are condition for those exception to throw. If you think that is not a good idea or a waste of time we can close this PR and i go on with other.

**Comment by karelz** [comment]
@stephentoub do you have general concerns about #23748? Or is it specific to this PR?

**Comment by stephentoub** [comment]
> do you have general concerns about #23748?

If an exception is used internally in a library, then making changes for it seems like unnecessary churn to adhere to design guidelines that are meant for public APIs.

If an exception isn't actually hit in code and is there along with an assert, potentially to satisfy the compiler, then changing it seems like unnecessary churn.

The only cases where it might be warranted is if an exception actually bubbles out through public API.  But there were hints earlier in the comment thread on this PR ("have you evaluated if these changes can change the exceptions being surfaced to the user?") that suggested to me we didn't want such public-facing changes.

**Comment by karelz** [comment]
Makes sense, thanks! (BTW: I didn't interpret it as we don't want public-facing changes here, but let's see what @saurabh500 will say :))

**Comment by saurabh500** [comment]
@karelz I agree with @stephentoub 
> If an exception is used internally in a library, then making changes for it seems like unnecessary churn to adhere to design guidelines that are meant for public APIs.

As a code reviewer, such changes are time consuming to validate in terms of figuring out if they are going to change what bubbles up to the customer. Also changing the type of exception for internal use warrants checking that all code paths which can catch this exception are still getting the right exception. 
The effort needed to validate this change may far outweigh the potential benefit of this change. (the benefit in this case is code that adheres to coding guidelines) 
Considering that the SqlClient code is very old, such changes don't add benefit for customers and in fact add churn for code reviewers, I would want to invest time on issues, fixing which would definitely make an impact to the customer experience with System.Data.* or .Net Core. I would welcome contributions / investigation in areas reported about functionality and performance improvements.

I do appreciate and respect @MarcoRossignoli effort in making these changes and holding high coding standards. If he is passionate about driving this, then I would want to see analysis about back compat, impact to any internal handling of these exceptions along with the rationale for why the exception types were changed to the types that are being proposed.

**Comment by MarcoRossignoli** [comment]
@karelz @saurabh500 @stephentoub if you agree with me for the moment we can close this PR and i can go on with other namespaces. As @karelz said on original issue "I think it would be best to discuss it per area", here seems that the effort/benefit are unbalanced. I'm newbie on contributing and my idea is to start with something "easy" to become familiar with project/dev tools. After some discussions and code analysis on this i think is better to me focus only on sure generic exceptions bubbles up to the customer(for instance on System.Drawing seems more clear). My idea is to help not to waste your precious time with useless code. Again thank's to all of you and sorry for my trivial change here.

**Comment by stephentoub** [comment]
Thanks, @MarcoRossignoli.  We do appreciate your interests in contributing here.  I'm sure we can find a good place where your efforts will be valuable.  I'll close this one out.

**Pull request closed without merging**